Content.Server/Cuffs/CuffableSystem.cs

```diff
@@ -39,5 +42,13 @@ private void OnCuffableGetState(EntityUid uid, CuffableComponent component, ref
             // the iconstate is formatted as blah-2, blah-4, blah-6, etc.
             // the number corresponds to how many hands are cuffed.
         }
+
+        private void OnBuckleAttemptEvent(EntityUid uid, CuffableComponent component, ref BuckleAttemptEvent args)
+        {
+            if (TryComp<HandsComponent>(uid, out var hands) && component.CuffedHandCount == hands.Count)
+            {
+                args.Cancelled = true;
+            }
```

Comment:
forgot to mention, this needs player feedback, make it make a popup or something telling you why you cant buckle/unbuckle (the message should change depending on the value of args.Buckling)